src/language/language_type.rs

```diff
@@ -19,23 +19,38 @@ impl LanguageType {
     /// Parses a given `Path` using the `LanguageType`. Returning `Stats`
     /// on success.
     pub fn parse(self, path: PathBuf) -> io::Result<Stats> {
+        self.parse_with_accuracy(path).map_err(|e| e.0)
+    }
+
+    /// Parses a given `Path` using the `LanguageType`. Returning `Stats`
+    /// on success and giving back ownership of PathBuf on error.
+    pub fn parse_with_accuracy(self, path: PathBuf) -> Result<Stats, (io::Error, PathBuf)> {
         let text = {
-            let f = File::open(&path)?;
+            let f = match File::open(&path) {
+                Ok(f) => f,
+                Err(e) => return Err((e, path)),
+            };
             let mut s = String::new();
             let mut reader = DecodeReaderBytes::new(f);
 
-            reader.read_to_string(&mut s)?;
+            if let Err(e) = reader.read_to_string(&mut s) {
+                return Err((e, path));
+            }
             s
         };
 
-        self.parse_from_str(path, &text)
+        Ok(self.parse_from_str_with_accuracy(path, &text))
     }
 
     /// Parses the text provided. Returning `Stats` on success.
-    pub fn parse_from_str(self, path: PathBuf, text: &str)
-        -> io::Result<Stats>
+    pub fn parse_from_str(self, path: PathBuf, text: &str) -> io::Result<Stats>
     {
+        Ok(self.parse_from_str_with_accuracy(path, text))
+    }
 
+    /// Parses the text provided.
+    pub fn parse_from_str_with_accuracy(self, path: PathBuf, text: &str) -> Stats
```

Comment:
Why is `parse_from_str_with_accuracy` necessary?

Reply:
oh it actually isnt now that you say it, i was just confused as to why `parse_from_str`returns a result when it never returns an error. Probably due to previous implementations I guess? Will remove the function again